# Section 3.6 Exercise 3

In [ ]:
# part a, find proportion of black defendents who received death penalty and white defendents who received death penalty

In [1]:
count <- c(53,414,11,37,0,16,4,139)
death <- factor(c(1,0,1,0,1,0,1,0))
defendant <- factor(c(0,0,1,1,0,0,1,1))
victim <- factor(c(0,0,0,0,1,1,1,1))
levels(death) <- c("no","yes")
levels(defendant) <- c("white","black")
levels(victim) <- c("white","black")

sum(count[death=="yes"&defendant=="black"])/
sum(count[defendant=="black"])

[1] 0.07853403

In [2]:
sum(count[death=="yes"&defendant=="white"])/
sum(count[defendant=="white"])

[1] 0.1097308

**7.85% of black defendents were sentenced to death while 10.97% of white defendents were sentenced to death**

In [ ]:
# part b, find the log-linear model that best explains these data

In [11]:
mod0 <- glm(count~death*defendant+death*victim+defendant*victim, family=poisson(link=log))
summary(mod0)
mod1 <-glm(count~death*victim+defendant*victim, family=poisson(link=log))
summary(mod1)
anova(mod0,mod1,test="Chisq")


Call:
glm(formula = count ~ death * defendant + death * victim + defendant * 
    victim, family = poisson(link = log))

Deviance Residuals: 
       1         2         3         4         5         6         7         8  
 0.02505  -0.00895  -0.05463   0.03000  -0.60362   0.04572   0.09251  -0.01545  

Coefficients:
                           Estimate Std. Error z value Pr(>|z|)    
(Intercept)                 6.02631    0.04913 122.669  < 2e-16 ***
deathyes                   -2.05946    0.14585 -14.121  < 2e-16 ***
defendantblack             -2.42032    0.17155 -14.109  < 2e-16 ***
victimblack                -3.26517    0.25478 -12.816  < 2e-16 ***
deathyes:defendantblack     0.86780    0.36707   2.364   0.0181 *  
deathyes:victimblack       -2.40444    0.60061  -4.003 6.25e-05 ***
defendantblack:victimblack  4.59497    0.31353  14.656  < 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for poisson family taken to be 1)

    Null de


Call:
glm(formula = count ~ death * victim + defendant * victim, family = poisson(link = log))

Deviance Residuals: 
       1         2         3         4         5         6         7         8  
-0.67084   0.24846   1.84218  -0.79292  -0.89724   0.10149   0.20843  -0.03411  

Coefficients:
                           Estimate Std. Error z value Pr(>|z|)    
(Intercept)                 6.01363    0.04916 122.323  < 2e-16 ***
deathyes                   -1.95258    0.13358 -14.618  < 2e-16 ***
victimblack                -3.26652    0.25511 -12.805  < 2e-16 ***
defendantblack             -2.27513    0.15157 -15.010  < 2e-16 ***
deathyes:victimblack       -1.70455    0.52373  -3.255  0.00114 ** 
victimblack:defendantblack  4.46538    0.30408  14.685  < 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 1225.080  on 7  degrees of freedom
Residual deviance:    5.394  on 2  degrees of free

Resid. Df Resid. Dev Df Deviance  Pr(>Chi)  
1 1         0.3798378  NA        NA         NA
2 2         5.3940390  -1 -5.014201 0.02514023

**The model in which counts of guilty verdicts depend on all two-way interactions is most appropriate. From mod0, the least significant interaction was between the defendant's skin color and whether or not the death penalty was received. However, leaving it out of the model (as shown by mod1) is not as appropriate because it causes the AIC to increase)**

In [ ]:
# part c, interpretation

Intercept:
The three independent variables are equal to zero when we have no death sentences, white victims, and white defendents. This means the expected number of guilty verdicts for no death sentences, white victim, and white defendent is exp(6.01363) =  408.96. This is the meaning of the intercept -- taking the exponential gives us the baseline number of deaths. 

On average, having a black victim is associated with a decrease of 3.27 guilty verdicts. On average, having a black victim decreases the number of guilty verdicts by exp(3.27) times compared to having a nonblack victim.

On average, assigning the death penalty to a black defendent is associated with a 0.867 increase in the number of guilty verdicts. On average, assigning the death penalty to a black defendent increases the number of guilty verdicts by exp(0.867) times compared to assigning the death penalty to a non-black defendent.**

On average, having a black victim and a black defendent is associated with a 4.465 increase in the number of guilty verdicts. On average, having a black victim and black defendent increases the number of guilty verdicts by exp(4.465) times compared to having a non-black victim and a non-black defendent.